sematic/db/models/resolution.py

```diff
@@ -0,0 +1,177 @@
+# Standard Library
+import uuid
+from enum import Enum, unique
+from typing import Dict, Optional, Union
+
+# Third party
+from sqlalchemy import Column, types
+from sqlalchemy.orm import validates
+
+# Sematic
+from sematic.db.models.base import Base
+from sematic.db.models.json_encodable_mixin import ENUM_KEY, JSONEncodableMixin
+
+
+@unique
+class ResolutionStatus(Enum):
+    """The status of the resolution itself.
+
+    This is distinct from the status of the root run: the root run might not be running
+    yet, but the resolution is. Or something may fail in the resolver itself, rather than
+    in one of the runs involved in it.
+
+    States
+    ------
+    CREATED:
+        The Resolution is in the DB, but k8s has not been asked to execute it yet.
+    SCHEDULED:
+        K8s (or the local process, for non-detached) has been asked to execute the
+        resolution, but the code has not started executing for it yet.
+    RUNNING:
+        The code for the resolution is executing.
+    FAILED:
+        There was an error in the resolution itself, NOT necessarily in the runs that it's
+        managing. The resolution may have started getting too many 500s from the Sematic
+        server, for example.
+    COMPLETE:
+        The resolution is done, and it will do no more work. This status may be used even
+        if the root run failed, so long as it failed due to some issue in the Sematic
+        func execution and not for some other reason. It may also be used if the root run
+        was canceled, so long as the cancellation was exited cleanly.
+    """
+
+    CREATED = "CREATED"
+    SCHEDULED = "SCHEDULED"
+    RUNNING = "RUNNING"
+    FAILED = "FAILED"
+    COMPLETE = "COMPLETE"
```

Comment:
Is this state machine required at this time? Can we get away with not adding this yet?

We could start with simply a data container for image sha, git commit, owner, but no state machine yet.